game-core/src/main/java/games/strategy/triplea/delegate/DiceRoll.java

```diff
@@ -1069,27 +1069,33 @@ private static int getSupport(
       final Predicate<UnitSupportAttachment> ruleFilter) {
     int givenSupport = 0;
     for (final List<UnitSupportAttachment> bonusType : supportsAvailable) {
+      final int bonusTypeCount = bonusType.get(0).getBonusType().getCount();
+      int maxPerBonusType = bonusTypeCount >= 1 ? bonusTypeCount : Integer.MAX_VALUE;
```

Comment:
I'm looking for ways this could be simpler, if you spot anything @ron-murhammer , please go for it.

Looking at these two lines, it's a bit odd as we converting a 'zero' value to a max integer. There seems to be quite a lot packed in these two lines.

I wonder if we can help this by encapsulating the 'infinite' knowledge into bonusType.

Perhaps we could have 'getBonusType().getCount()' contain the logic directly and have that method convert the '0' to be a 'max_value'?

In which case we would have:

```
      int maxPerBonusType = bonusType.get(0).getBonusType().getCount();;
```

That also dovetails nicely with having an explicit flag for 'infinite', the bonus type can take care of that detail without 'DiceRoll' knowing about the details of unit bonus allotment (arguably inappropriate coupling).

Reply:
Agree. Updated.